app/controllers/application_controller.rb

```diff
@@ -5,6 +5,146 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "unbreakable_password"
   end
 
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in? 
+      redirect '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    user = User.new(params) 
+    if !user.attributes.values.include?("")&& user.save
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/login' do
+    if logged_in? 
+      redirect '/tweets'
+    else
+      erb :'/users/login'
+    end  
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect '/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect '/login'
+    else
+      redirect '/'
+    end
+  end
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    erb :'/users/user_show'
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = User.find(session[:user_id])
+      erb :'/tweets/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if !logged_in?
+      redirect '/tweets/new'
+    elsif  params[:content].empty? 
+      redirect '/tweets/new'
+    else
+      user = current_user
+      tweet = Tweet.create(content: params[:content], user_id: user.id)
+      redirect '/tweets'
+    end
+  end
+
+  get '/tweets/:id' do
+    if !logged_in?
+      redirect '/login'
+    else
+      @tweet = Tweet.find(params[:id])
+      erb :'/tweets/show_tweet'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == current_user.id
+       erb :'tweets/edit_tweet'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end 
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find(params[:id])
+      @tweet.update(content: params[:content])
+      @tweet.save
```

Comment:
@tweet.update already returns a `true` or `false` value and commits to the database, so `.save` is redundant.